libsolidity/codegen/ir/IRGeneratorForStatements.h

```diff
@@ -112,7 +112,8 @@ class IRGeneratorForStatements: public ASTConstVisitor
 
 	/// @returns a Yul expression representing the current value of @a _expression,
 	/// converted to type @a _to if it does not yet have that type.
-	std::string expressionAsType(Expression const& _expression, Type const& _to);
+	/// If @a _forceCleanup is set to true, it also cleans the value, in case it already has type @a _to.
+	std::string expressionAsType(Expression const& _expression, Type const& _to, bool _forceCleanup = false);
```

Comment:
Are the more occurrences of "no cleanup wanted" than "cleanup wanted"?

Reply:
So far it didn't clean up... and the only place we need the cleaning version right now is for the comparison operations... although it might make sense to look at all occurrences of ``expressionAsType`` and check whether we don't *actually* want cleanups there as well...

Reply:
Ok!